common/transport/amqp/devdoc/receiver_link_requirements.md

```diff
@@ -0,0 +1,98 @@
+
+```typescript
+import * as amqp10 from 'amqp10';
+import { AmqpMessage } from './amqp_message';
+
+const linkAddress = 'exampleAddress';
+const amqp10Client = new amqp10.AmqpClient(null);
+
+const receiverLink = new ReceiverLink(linkAddress, null, amqp10Client);
+receiverLink.on('error', (err) => {
+  console.error(err.toString());
+});
+
+receiverLink.on('message', (msg) => {
+  console.log(msg.body);
+  receiverLink.accept(msg, (err) => {
+    if (err) {
+      console.error(err.toString());
+    } else {
+      console.log('message successfully settled (accept)');
+    }
+  });
+});
+```
+
+## Public Interface
+
+### constructor(linkAddress: string, linkOptions: any, amqp10Client: amqp10.AmqpClient)
+
+**SRS_NODE_AMQP_RECEIVER_LINK_16_002: [** The `ReceiverLink` class shall inherit from `EventEmitter`. **]**
+
+**SRS_NODE_AMQP_RECEIVER_LINK_16_003: [** The `ReceiverLink` class shall implement the `AmqpLink` interface. **]**
+
+### attach(callback: (err?: Error) => void): void
+
+**SRS_NODE_AMQP_RECEIVER_LINK_16_004: [** The `attach` method shall use the stored instance of the `amqp10.AmqpClient` object to attach a new link object with the `linkAddress` and `linkOptions` provided when creating the `ReceiverLink` instance. **]**
+
+**SRS_NODE_AMQP_RECEIVER_LINK_16_006: [** The `ReceiverLink` object should subscribe to the `detached` event of the newly created `amqp10` link object. **]**
+
+**SRS_NODE_AMQP_RECEIVER_LINK_16_007: [** The `ReceiverLink` object should subscribe to the `errorReceived` event of the newly created `amqp10` link object. **]**
+
+**SRS_NODE_AMQP_RECEIVER_LINK_16_008: [** If the `amqp10.AmqpClient` fails to create the link the `callback` function shall be called with this error object. **]**
+
+**SRS_NODE_AMQP_RECEIVER_LINK_16_020: [** The `attach` method shall call the `callback` if the link was successfully attached **]**
+
+### detach(): void
+
+**SRS_NODE_AMQP_RECEIVER_LINK_16_009: [** The `detach` method shall detach the link created by the `amqp10.AmqpClient` underlying object. **]**
+
+### accept(message: AmqpMessage, callback: (err?: Error, result?: results.MessageAccepted) => void): void
+
+**SRS_NODE_AMQP_RECEIVER_LINK_16_021: [** The `accept` method shall throw if the `message` argument is falsy. **]**
+
+**SRS_NODE_AMQP_RECEIVER_LINK_16_013: [** The `accept` method shall use the link created by the underlying `amqp10.AmqpClient` to settle the specified `message` with IoT hub by accepting it. **]**
+
+**SRS_NODE_AMQP_RECEIVER_LINK_16_022: [** The `accept` method shall work whether a `callback` is specified or not, and call the callback with a `result.MessageCompleted` object if a callback is specified. **]**
+
+**SRS_NODE_AMQP_RECEIVER_LINK_16_014: [** If the state machine is not in the `attached` state, the `accept` method shall immediately fail with a `DeviceMessageLockLostError`. **]**
+
+### complete(message: AmqpMessage, callback: (err?: Error, result?: results.MessageAccepted) => void): void
```

Comment:
>callback [](start = 35, length = 8)

optional?

Reply:
yes

---
In reply to: [139493516](https://github.com/Azure/azure-iot-sdk-node/pull/83#discussion_r139493516) [](ancestors = 139493516)